### PR TITLE
Health care contract improvements

### PR DESCRIPTION
**Pull request by LEO1612D**
This PR includes improvements in current healthcare contract,
- Added custom modifiers to validate data before calling function.
- Improve require by adding failure message.

Please verify changes before merging, thanks.

### CODE REVIEW

**Comment by LEO1612D** [comment]
@Rishabh42 I Updated the changes, 
- complied contract on remix after changes => looks good.
- updated deploy method.

Please review these changes, and let me know if there is anything.

File: migrations/2_deploy_HealthCare.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var HealthCare = artifacts.require("./HealthCare.sol");
+const TO_LAB_ADMIN_ADDRESS = "0xF6F2F51c07e44efE7BC25E0EC6B332f39d930ac0";
```
Comment (Rishabh42) [inline migrations/2_deploy_HealthCare.js:2]: Pls add a comment in front of this, something like `hard coded address from Ganache`

**Review by Rishabh42 — CHANGES_REQUESTED** [review]
Looks good @LEO1612D , please review my comment.
And would be great if you could also update the README, specifically step 3 since we're not hard coding that address in the smart contract

**Comment by LEO1612D** [comment]
@Rishabh42 Done.

File: README.md [hunk 2]
```diff
@@ -6,39 +6,39 @@ Problem statement:
...
-2. Run an instance of ganache-cli on your local machine and connect your metamask wallet to it. Also, add the first 3 accounts from ganache to your metamask by importing their private keys and assign the following names to it:  
-    account 1: Hospital admin  
-    account 2: Lab admin  
-    account 3: Patient  
-3. Replace the Lab admin's address inside the constructor on line 27 with the Lab admin's address that we configured in the previous step.
+2. Run an instance of ganache-cli on your local machine and connect your metamask wallet to it. Also, add the first 3 accounts from ganache to your metamask by importing their private keys and assign the following names to it:
+    account 1: Hospital admin
+    account 2: Lab admin
+    account 3: Patient
+3. Replace the Lab admin's address [here in file : 2_deploy_HealthCare.js](https://github.com/Rishabh42/HealthCare-Insurance-Ethereum/blob/60792a7d275e84cca7aa3a6d4734554cdbd67283/migrations/2_deploy_HealthCare.js#L2) with the Lab admin's address that we configured in the previous step.
```
Comment (Rishabh42) [inline README.md:18]: These steps are for deploying on Remix, so it should be that the user should pass an address from Ganache (which is different from the hospital admin's address) to the constructor while deploying the contract.

**Review by Rishabh42 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (LEO1612D) [reply README.md:18]: Pass The Lab admin's address (From ganache) in constructor [here in file : 2_deploy_HealthCare.js](https://github.com/Rishabh42/HealthCare-Insurance-Ethereum/blob/60792a7d275e84cca7aa3a6d4734554cdbd67283/migrations/2_deploy_HealthCare.js#L2).

-- this works ?

**Review by Rishabh42 — CHANGES_REQUESTED** [review]

**Comment by LEO1612D** [comment]
> Merging it, thanks @LEO1612D !

Thanks !! looking forward to contributing more 💯

**Pull request merged**